File: Homework1/C#/WinFormsAppHomeW1/WinFormsAppHomeW1/Form1.cs
namespace WinFormsAppHomeW1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if(richTextBox4.BackColor == Color.White)
            {
                richTextBox1.BackColor = Color.Green;
                richTextBox4.BackColor = Color.Green;
                richTextBox2.BackColor = Color.White;
                richTextBox4.Text = "Green Light";
            }
            else if (richTextBox4.BackColor == Color.Red)
            {
                richTextBox1.BackColor = Color.Green;
                richTextBox4.BackColor = Color.Green;
                richTextBox3.BackColor = Color.White;
                richTextBox2.BackColor = Color.White;
                richTextBox4.Text = "Green Light";
            }

            else if (richTextBox4.BackColor == Color.Green)
            {
                richTextBox3.BackColor = Color.Red;
                richTextBox4.BackColor = Color.Red;
                richTextBox1.BackColor = Color.White;
                richTextBox2.BackColor = Color.White;
                richTextBox4.Text = "Red Light";
            }
        }

        private void Button1_MouseHover(object sender, EventArgs e)
        {
            if(richTextBox1.BackColor == Color.Green || richTextBox3.BackColor == Color.Red)
            {
                richTextBox2.BackColor = Color.Yellow;
                richTextBox1.BackColor = Color.White;
                richTextBox3.BackColor = Color.White;
            }
        }

        private void Button1_MouseLeave(object sender, EventArgs e)
        {
            if(richTextBox2.BackColor == Color.Yellow)
            {
                richTextBox2.BackColor = Color.White;
                richTextBox1.BackColor = Color.White;
                richTextBox3.BackColor = Color.White;
                richTextBox4.BackColor = Color.White;
                richTextBox4.Text = "";
            }
        }
    }
}

File: Homework1/VB/WinFormsAppHVBnet1/WinFormsAppHVBnet1/Form1.vb
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If (RichTextBox4.BackColor = Color.White) Then
            RichTextBox1.BackColor = Color.Green
            RichTextBox4.BackColor = Color.Green
            RichTextBox2.BackColor = Color.White
            RichTextBox4.Text = "Green Light"
        ElseIf (RichTextBox4.BackColor = Color.Red) Then
            RichTextBox1.BackColor = Color.Green
            RichTextBox4.BackColor = Color.Green
            RichTextBox3.BackColor = Color.White
            RichTextBox2.BackColor = Color.White
            RichTextBox4.Text = "Green Light"
        ElseIf (RichTextBox4.BackColor = Color.Green) Then
            RichTextBox3.BackColor = Color.Red
            RichTextBox4.BackColor = Color.Red
            RichTextBox1.BackColor = Color.White
            RichTextBox2.BackColor = Color.White
            RichTextBox4.Text = "Red Light"
        End If
    End Sub

    Private Sub Button1_MouseHover(sender As Object, e As EventArgs) Handles Button1.MouseHover
        If (RichTextBox1.BackColor = Color.Green Or RichTextBox3.BackColor = Color.Red) Then
            RichTextBox2.BackColor = Color.Yellow
            RichTextBox1.BackColor = Color.White
            RichTextBox3.BackColor = Color.White
        End If
    End Sub

    Private Sub Button1_MouseLeave(sender As Object, e As EventArgs) Handles Button1.MouseLeave
        If (RichTextBox2.BackColor = Color.Yellow) Then
            RichTextBox2.BackColor = Color.White
            RichTextBox1.BackColor = Color.White
            RichTextBox3.BackColor = Color.White
            RichTextBox4.BackColor = Color.White
            RichTextBox4.Text = ""
        End If
    End Sub
End Class
